OpenRA.Mods.Common/Widgets/Logic/SettingsLogic.cs

```diff
@@ -216,7 +216,11 @@ Action InitDisplayPanel(Widget panel)
 			colorPreview.Color = ps.Color;
 
 			var colorDropdown = panel.Get<DropDownButtonWidget>("PLAYERCOLOR");
-			colorDropdown.OnMouseDown = _ => ColorPickerLogic.ShowColorDropDown(colorDropdown, colorPreview, worldRenderer.World);
+			if (worldRenderer.World.Type == WorldType.Shellmap)
+				colorDropdown.OnMouseDown = _ => ColorPickerLogic.ShowColorDropDown(colorDropdown, colorPreview, worldRenderer.World);
+			else
+				colorDropdown.Disabled = true;
```

Comment:
This won't change it back. 

Just add `colorDropdown.IsDisabled = () => worldRenderer.World.Type =! WorldType.Shellmap;` instead of the above.